### ISSUE TITLE
Neo4JClientFactory does not use externalProperties

### ISSUE THREAD

**Issue by danielfrg** [Impetus/kundera#541]
I am trying to overwrite the default `persistence.xml` settings using: `emf = Persistence.createEntityManagerFactory(db, kunderaMap);`, works fine for HBase but not for Neo4J.

I debug it a little bit and `Neo4JClientFactory.createPoolOrConnection()` is only using the persistenceUnit settings not the `externalProperties` attribute. Like this: `String datastoreFilePath = (String) props.get(PersistenceProperties.KUNDERA_DATASTORE_FILE_PATH);`

Comparing to `HBaseClientFactory` it has stuff like: `externalProperty.get(PersistenceProperties.KUNDERA_NODES);`. So maybe is an easy fix.

**Comment by mevivs** [comment]
Looks like an issue. Planning to contribute a pull request for this? 

-Vivek

**Comment by danielfrg** [comment]
I tried using a different Factory based on `Neo4JClientFactory` doing something like: `<property name="kundera.client.lookup.class" value="MyFactory" />` and overwriting the `createPoolOrConnection()` method but couldn't make it work (not a Java expert). I ended up fixing it using different persistence units since it was faster for the tests I am doing, i don't like it but is working.

If I got some time I will try to setup a better environment and try to fix it directly, can't promise anything though.
